bin/unit_metrics.py

```diff
@@ -42,32 +43,33 @@
 
 
 def application(environ, start_response):
-    connection = http.client.HTTPConnection("localhost:8181")
-    connection.request("GET", "/status")
-    response = connection.getresponse()
+    retries = 5
+    try:
+        connection = http.client.HTTPConnection("localhost:8181")
+        connection.request("GET", "/status")
+        response = connection.getresponse()
 
-    statj = json.loads(response.read())
-    connection.close()
+        statj = json.loads(response.read())
+        connection.close()
```

Comment:
I think we will need to ensure the connection is closed on the error path too, though I'm not sure how Python handles these on an exception

Reply:
true - needs to move to `finally` block